### PR TITLE
Member

### PR DESCRIPTION
**Pull request by olenashevchuk**
Member components: List, Select, Combined, Views, Forms. I am looking forward to get impoved Select and Combined

### CODE REVIEW

File: src/app/constants/index.js [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 import * as currency from './currency.js'
 import { route } from './route.js'
 import { color } from './color.js'
-import { role } from './role.js'
+import ROLE, { ROLE_KEYS, ROLE_VALUES } from './role.js'
 import { priority } from './priority.js'
-export { firebaseConfig, currency, route, color, role, priority }
+
+export {
+  firebaseConfig,
```
Comment (alexpassss) [inline src/app/constants/index.js:10]: all constants should be uppercased

File: src/app/constants/index.js [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
+  db,
```
Comment (alexpassss) [inline src/app/constants/index.js:11]: I saw the same in other pull requests

**Review by alexpassss — CHANGES_REQUESTED** [review]
please refactor your code

File: src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+  Snackbar,
+  FormHelperText
+} from '@material-ui/core'
+import { useForm, Controller } from 'react-hook-form'
+import { Alert } from '@material-ui/lab'
+import AddIcon from '@material-ui/icons/Add'
+import { RoleSingleSelect } from 'app/domains/Role/components/select'
+import { db } from 'app/constants'
+
+var md5 = require('md5')
```
Comment (VladBogach) [inline src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js:21]: use import unstead of require

File: src/app/constants/role.js [hunk 4]
```diff
@@ -1,2 +1,7 @@
-const role = {}
-export { role }
+const ROLE = { admin: 'admin', user: 'user', observer: 'observer' }
+
+const ROLE_KEYS = Object.keys(ROLE)
+const ROLE_VALUES = Object.values(ROLE)
+
+export default ROLE
+export { ROLE_KEYS, ROLE_VALUES }
```
Comment (ARTfun00) [inline src/app/constants/role.js:7]: Could u explain what's the difference between the ROLE_KEYS and ROLE_VALUES ?)

File: src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js [hunk 5]
```diff
@@ -1,8 +1,233 @@
...
+import PropTypes from 'prop-types'
+import { useForm, Controller } from 'react-hook-form'
+import { RoleSingleSelect } from 'app/domains/Role/components/select'
+import { Container, Row, Col } from '@qonsoll/react-design'
+import { expensesProject } from '../../../../../constants'
+
 const MemberAdvancedForm = (props) => {
-  return <div>MemberAdvancedForm</div>
+  const { handleSubmit, control, register, errors } = useForm({
+    defaultValues: { ...props }
```
Comment (ARTfun00) [inline src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js:24]: This line of code could be simplified to the `defaultValues: props`

File: src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js [hunk 6]
```diff
@@ -1,8 +1,233 @@
...
+  })
+  const fileUpload = useRef(null)
+
+  const [date, setDate] = useState(props.date ? props.date : '1609855440000')
```
Comment (ARTfun00) [inline src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js:28]: Something really strange is happening here, just think about: state variable called _date_, but u initiate it with long number, which actually has string type... please rename the state variable name or use appropriate initial and future values...

File: src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js [hunk 7]
```diff
@@ -1,8 +1,233 @@
...
+  const [avatar, setAvatar] = useState(props.avatar)
```
Comment (ARTfun00) [inline src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js:29]: Maybe it's avatarUrl ?)

File: src/app/index.js [hunk 8]
```diff
@@ -1,17 +1,17 @@
+import { Box } from '@material-ui/core'
 import { Navbar } from 'components'
 import { MemberShow } from 'domains/Member/routes'
 import { LogAll } from 'app/domains/Log/routes'
+import { MemberAdvancedForm } from 'app/domains/Member/components/forms'
 
 import { firebaseConfig } from 'app/constants'
 
 const App = () => {
   console.log(firebaseConfig)
```
Comment (ARTfun00) [inline src/app/index.js:10]: Please remove unnecessary console.log()-s

File: src/app/styles/bootstrap-grid-override.css [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+.col-xl-9,
+.col-xl-auto {
+  padding-left: 16px;
+  padding-right: 16px;
+}
+
+.row {
+  margin-left: -16px;
+  margin-right: -16px;
+}
```
Comment (ARTfun00) [inline src/app/styles/bootstrap-grid-override.css:80]: No newline at the end of the file. Have u configured and used linter ?

**Review by ARTfun00 — COMMENTED** [review]
Committing and pushing changes more frequently will prevent suffering and anger of reviewers... Please, make your pull-request smaller for the next times =)

**Comment by olenashevchuk** [comment]
-renamed prop avatar
-renamed constants
-replace styles to other files
-added component avatarUploader

File: src/app/components/Lib/Avatar/Avatar/Avatar.template.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <Avatar className={size} alt={props.alt} src={props.src}>
+      {props.children}
+    </Avatar>
+  )
+}
+
+CustomAvatar.propTypes = {
+  size: PropTypes.oneOf(['sm', 'md', 'lg'])
```
Comment (alexpassss) [inline src/app/components/Lib/Avatar/Avatar/Avatar.template.js:20]: Move sizes to the constants

File: src/app/components/Lib/Avatar/AvatarUploader/AvatarUploader.template.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+import { Row, Col } from '@qonsoll/react-design'
+import { Avatar } from '../Avatar'
+import { EXPENSES_PROJECT } from 'app/constants'
+import styles from './AvatarUploader.styles'
+import { makeStyles } from '@material-ui/core/styles'
+
+const useStyles = makeStyles(styles)
+
+const CustomAvatarUploader = (props) => {
+  const { onChange, value } = props
```
Comment (alexpassss) [inline src/app/components/Lib/Avatar/AvatarUploader/AvatarUploader.template.js:14]: Add comments

File: src/app/components/Lib/Avatar/AvatarUploader/AvatarUploader.template.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    await fileRef.put(fileUpload.current.files[0])
+    fileRef.getDownloadURL().then((url) => {
+      setAvatarUrl(url)
+      onChange(url)
+    })
+  }
+
+  const classes = useStyles()
+  return (
+    <>
```
Comment (alexpassss) [inline src/app/components/Lib/Avatar/AvatarUploader/AvatarUploader.template.js:36]: Container should be first element

File: src/app/constants/index.js [hunk 13]
```diff
@@ -1,7 +1,21 @@
...
+import ROLE, { ROLE_KEYS, ROLE_VALUES } from './role.js'
 import { priority } from './priority.js'
-export { firebaseConfig, currency, route, color, role, priority }
+
+export {
+  FIREBASE_CONFIG,
+  STORE,
+  STORAGE,
+  EXPENSES_PROJECT,
+  currency,
```
Comment (alexpassss) [inline src/app/constants/index.js:14]: Plural/singular
Uppercase

File: src/app/domains/Color/components/select/ColorSingleSelect/ColorSingleSelect.template.js [hunk 14]
```diff
@@ -1,5 +1,33 @@
+import { useState } from 'react'
+import MenuItem from '@material-ui/core/MenuItem'
+import Select from '@material-ui/core/Select'
+import { makeStyles } from '@material-ui/core/styles'
+
+import { COLOR_VALUE } from '../../../../../constants'
+
+const useStyles = makeStyles((theme) => ({
```
Comment (alexpassss) [inline src/app/domains/Color/components/select/ColorSingleSelect/ColorSingleSelect.template.js:8]: Please move to the separate file

File: src/app/domains/Color/components/select/ColorSingleSelect/ColorSingleSelect.template.js [hunk 15]
```diff
@@ -1,5 +1,33 @@
...
+    textAlign: 'center'
+  }
+}))
+
 const ColorSingleSelect = (props) => {
-  return <div>ColorSingleSelect</div>
+  const classes = useStyles()
+  const [color, setColor] = useState(COLOR_VALUE[0].name)
+  return (
+    <Select className={classes.selectColor} value={color}>
```
Comment (alexpassss) [inline src/app/domains/Color/components/select/ColorSingleSelect/ColorSingleSelect.template.js:19]: What about universal select?

File: src/app/domains/Currency/components/views/CurrencySimpleView/CurrencySimpleView.template.js [hunk 16]
```diff
@@ -1,24 +1,8 @@
...
-CurrencySimpleView.propTypes = {
-  value: PropTypes.oneOf(CURRENCY_KEYS).isRequired,
-  type: PropTypes.oneOf(['sign', 'cc'])
-}
-CurrencySimpleView.defaultProps = {
-  type: 'sign',
-  value: 'USD'
-}
+CurrencySimpleView.propTypes = {}
+CurrencySimpleView.defaultProps = {}
```
Comment (alexpassss) [inline src/app/domains/Currency/components/views/CurrencySimpleView/CurrencySimpleView.template.js:6]: Remove default props everywhere

File: src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js [hunk 17]
```diff
@@ -0,0 +1,142 @@
...
+  DialogTitle,
+  Fab,
+  Snackbar,
+  FormHelperText
+} from '@material-ui/core'
+import { useForm, Controller } from 'react-hook-form'
+import { Alert } from '@material-ui/lab'
+import AddIcon from '@material-ui/icons/Add'
+import { RoleSingleSelect } from 'app/domains/Role/components/select'
+import { STORE } from '../../../../../constants'
```
Comment (alexpassss) [inline src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js:19]: Is it possible to simplify path ?

File: src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js [hunk 18]
```diff
@@ -0,0 +1,142 @@
...
+import md5 from 'md5'
+
+const MemberCombined = (props) => {
+  const [open, setOpen] = React.useState(false)
+  const [openSnackbarSuccess, setOpenSnackbarSuccess] = React.useState(false)
+  const [openSnackbarError, setOpenSnackbarError] = React.useState(false)
+  const { register, handleSubmit, control, errors } = useForm()
+
+  const onSubmit = (data) => {
```
Comment (alexpassss) [inline src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js:28]: What about universal save?

File: src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js [hunk 19]
```diff
@@ -0,0 +1,142 @@
...
+        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
+        open={openSnackbarError}
+        autoHideDuration={6000}
+        onClose={handleClose}>
+        <Alert variant="filled" severity="error">
+          This is an error message!
+        </Alert>
+      </Snackbar>
+
+      <Dialog
```
Comment (alexpassss) [inline src/app/domains/Member/components/combined/MemberCombined/MemberCombined.template.js:78]: Universal dialog?

File: src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js [hunk 20]
```diff
@@ -1,7 +1,182 @@
...
+    defaultValues: { ...props }
+  })
+
+  const [dateInSeconds, setDateInSeconds] = useState(
+    props.dateInSeconds ? props.dateInSeconds : 946677600000
+  )
 
+  const onSubmit = (data) => {
+    data.dateInSeconds = dateInSeconds
+    console.log(data)
```
Comment (alexpassss) [inline src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js:31]: Remove console.log

File: src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js [hunk 21]
```diff
@@ -1,7 +1,182 @@
...
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <KeyboardDatePicker
+                fullWidth
+                margin="normal"
+                id="date-picker-dialog"
+                label="Date picker dialog"
+                name="date"
+                format="dd/MM/yyyy"
+                inputValue={
+                  new Date(dateInSeconds).getDate() +
```
Comment (alexpassss) [inline src/app/domains/Member/components/forms/MemberAdvancedForm/MemberAdvancedForm.template.js:118]: Use moment.js instead

File: src/app/domains/Member/components/list/MemberList/MemberList.template.js [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+            fullWidth
+            placeholder="Search..."
+            id="input-with-icon-adornment"
+            startAdornment={
+              <InputAdornment position="start">
+                <SearchIcon />
+              </InputAdornment>
+            }
+          />
+          <List>
```
Comment (alexpassss) [inline src/app/domains/Member/components/list/MemberList/MemberList.template.js:34]: Universal list should be here

**Review by alexpassss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**